spec/models/post_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+Rspec.describe Post, type: :model do
+  subject { Post.new(title: 'Welcome', author: User.create(name: 'John')) }
+
+  before { subject.save }
+
+  describe 'validate test' do
+    it 'title should be present' do
+      subject.title = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'title should be less than 251 characters' do
+      subject.title = 'Lorem ipsum dolor sit amet,
+      consectetuer adipiscing elit. Aenean commodo ligula
+      eget dolor. Aenean massa. Cum sociis natoque penatibus
+      et magnis dis parturient montes, nascetur ridiculus mus.
+      Donec quam felis, ultricies neln ea' # 251 chars
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments_counter should be integer' do
+      subject.comments_counter = 'subject'
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments_counter should be greater than or equal to zero' do
+      subject.comments_counter = -2
+      expect(subject).to_not be_valid
+
+      subject.comments_counter = 0
+      expect(subject).to be_valid
+    end
+
+    it 'likes_counter should be integer' do
+      subject.likes_counter = 'subject'
+      expect(subject).to_not be_valid
+    end
+
+    it 'likes_counter should be greater than or equal to zero' do
+      subject.likes_counter = -2
+      expect(subject).to_not be_valid
+
+      subject.likes_counter = 0
+      expect(subject).to be_valid
+    end
+  end
+
+  describe '#update_user_posts_counter' do
+    it 'updates the user posts_counter attribute' do
+      # Arrange
+      user = User.create(name: 'John')
+      post = Post.create(title: 'Hello', author: user)
+
+      # Act
+      post.update_user_posts_counter
+
+      # Assert
+      expect(user.reload.posts.counter).to eq(1)
```

Comment:
* Kindly fix this error I get check screenshot below 👇 . HINT 💭 if you check your post model you will see that the post counts are assign to a variable `posts_counter` so therefore the correct line of code in your test is `expect(user.reload.posts_counter).to eq(1)`
![image](https://github.com/unorjikingsley/ROR-BlogApp/assets/52497246/e0c63d66-d40f-4720-af35-d2fb9d602660)